### PR TITLE
Fix: Use DocumentRendererInterface instead DocumentRenderer

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
Noticed it by https://github.com/pimcore/pimcore/pull/16216

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 23100cc</samp>

This pull request refactors the document rendering logic in various classes and services to use a new interface `DocumentRendererInterface` instead of the old service `DocumentRenderer`. This improves the flexibility and consistency of the document rendering process and deprecates the old service. The upgrade documentation is updated accordingly.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 23100cc</samp>

> _`DocumentRenderer`_
> _Deprecated, use interface_
> _Spring cleaning the code_

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 23100cc</samp>

*  Introduce a new interface `DocumentRendererInterface` and deprecate the old service `DocumentRenderer` ([link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-20cadf787aa9b8e9f281fcc3077ad46d5027868776371979c21a6ccf5f4e2fcbL54-R64), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-375aded12dfb868b84d5b8c27d1aed2358680a27b683e8a985f1b85193dd79aaR9-R11))
* Replace the usage of the old service `DocumentRenderer` with the new interface `DocumentRendererInterface` in the constructors and methods of the following classes: `DocumentFallbackHandler`, `ResponseExceptionListener`, `EditableUsageResolver`, `StaticPageGenerator`, and `Service` ([link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-20cadf787aa9b8e9f281fcc3077ad46d5027868776371979c21a6ccf5f4e2fcbL63-R73), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-083637c221d9fe6b3a2616d0115697a08baf188419036a6406f5509aa1998f0eL21-R21), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-083637c221d9fe6b3a2616d0115697a08baf188419036a6406f5509aa1998f0eL45-R45), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-4dd848043bdc03054af545b20ba713afd0b55bc1e8a00e3493eaf466e31b0f8bL20-R20), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-4dd848043bdc03054af545b20ba713afd0b55bc1e8a00e3493eaf466e31b0f8bL35-R37), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-d457906ebd50f5e7051931d821a79d12970959978069590f47ee8f203de98d67L20-R20), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-d457906ebd50f5e7051931d821a79d12970959978069590f47ee8f203de98d67L31-R31), [link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-be2f2234588c6da18c7b5b84a009810bbe0efe6be173d1b2a219147eb0966794L74-R73))
* Remove the unused use statement of the old service `DocumentRenderer` from the `Service` class ([link](https://github.com/pimcore/pimcore/pull/16267/files?diff=unified&w=0#diff-be2f2234588c6da18c7b5b84a009810bbe0efe6be173d1b2a219147eb0966794L20))

### CODE REVIEW

File: lib/Document/StaticPageGenerator.php [hunk 1]
```diff
@@ -28,8 +28,11 @@
 
 class StaticPageGenerator
 {
-    public function __construct(protected DocumentRenderer $documentRenderer, private LockFactory $lockFactory, protected SystemSettingsConfig $settingsConfig)
-    {
+    public function __construct(
+        protected DocumentRendererInterface $documentRenderer,
```
Comment (brusch) [inline lib/Document/StaticPageGenerator.php:32]: That's unfortunately a BC-break, as this class is not `@internal` 😬

**Review by brusch — CHANGES_REQUESTED** [review]

File: lib/Document/StaticPageGenerator.php (hunk 1 above)
Comment (blankse) [reply lib/Document/StaticPageGenerator.php:32]: Why? When someone overwrite this class it can still have the old cunstruct and it works.

Comment (jdreesen) [reply lib/Document/StaticPageGenerator.php:32]: That's right, contravariance does not affect constructors.

Comment (brusch) [reply lib/Document/StaticPageGenerator.php:32]: 👍 Thanks for clarifying guys - was not entirely sure about that one 😊

**Pull request merged**